tools/c7n_azure/c7n_azure/resources/front_door.py

```diff
@@ -70,5 +71,58 @@ def process(self, resources, event=None):
                 front_endpoint = client.frontend_endpoints.get(
                     front_door['resourceGroup'], front_door['name'],front_endpoints['name'])
                 if self.check_state(front_endpoint.web_application_firewall_policy_link):
+                    # what if one front_door has multiple endpoints that match?
```

Comment:
```python
if front_door not in matched:
    matched.append(front_door)
```

for small cardinalities this works fine.

Reply:
We can dispense with additional `GET` requests because front door's json contains information about each endpoint's link:

**Disabled:**
```json
{
  "frontendEndpoints": [
    {
      "id": "/subscriptions/dd3ce4b8-093a-40da-8369-d50149d0f84a/resourcegroups/FrontDoorExampleResourceGroup/providers/Microsoft.Network/Frontdoors/42342342-FrontDoor/FrontendEndpoints/exampleFrontendEndpoint1",
      "name": "exampleFrontendEndpoint1",
      "type": "Microsoft.Network/Frontdoors/FrontendEndpoints",
      "properties": {
        "hostName": "42342342-FrontDoor.azurefd.net",
        "sessionAffinityEnabledState": "Disabled",
        "sessionAffinityTtlSeconds": 0,
        "resourceState": "Enabled"
      }
    }
  ]
}

```
**Enabled:**
```json
{
  "frontendEndpoints": [
    {
      "id": "/subscriptions/dd3ce4b8-093a-40da-8369-d50149d0f84a/resourcegroups/FrontDoorExampleResourceGroup/providers/Microsoft.Network/Frontdoors/42342342-FrontDoor/FrontendEndpoints/exampleFrontendEndpoint1",
      "name": "exampleFrontendEndpoint1",
      "type": "Microsoft.Network/Frontdoors/FrontendEndpoints",
      "properties": {
        "hostName": "42342342-FrontDoor.azurefd.net",
        "sessionAffinityEnabledState": "Disabled",
        "sessionAffinityTtlSeconds": 0,
        "webApplicationFirewallPolicyLink": {
          "id": "/subscriptions/dd3ce4b8-093a-40da-8369-d50149d0f84a/resourceGroups/FrontDoorExampleResourceGroup/providers/Microsoft.Network/frontdoorWebApplicationFirewallPolicies/examplefdwafpolicy"
        },
        "resourceState": "Enabled"
      }
    }
  ]
}
```
<img width="1413" alt="image" src="https://github.com/cloud-custodian/cloud-custodian/assets/57749020/844d15e0-22c7-4545-b7b0-b500b1491f10">

I pushed some fixes, review pls !